src/main/java/cn/nukkit/event/block/AnvilDamageEvent.java

```diff
@@ -1,45 +1,167 @@
+)
+@Since("1.1.1.0-PN")
 public class AnvilDamageEvent extends BlockFadeEvent implements Cancellable {
     
     private static final HandlerList handlers = new HandlerList();
-    
+
+    @Since("1.1.1.0-PN")
     public static HandlerList getHandlers() {
         return handlers;
     }
     
     private final Player player;
     private final CraftingTransaction transaction;
-    private final Cause cause;
-    
-    public AnvilDamageEvent(Block block, Block newState, Player player, CraftingTransaction transaction, Cause cause) {
+    private final DamageCause cause;
+
+    @Since("1.3.2.0-PN")
+    public AnvilDamageEvent(Block block, int oldDamage, int newDamage, DamageCause cause, Player player) {
+        this(adjustBlock(block, oldDamage), adjustBlock(block, newDamage), player, null, cause);
+    }
+
+    @PowerNukkitOnly
+    @Since("1.3.2.0-PN")
+    public AnvilDamageEvent(Block block, Block newState, Player player, @Nullable CraftingTransaction transaction, DamageCause cause) {
         super(block, newState);
         this.player = player;
         this.transaction = transaction;
         this.cause = cause;
     }
     
+    @PowerNukkitOnly
+    @Since("1.1.1.0-PN")
+    public AnvilDamageEvent(Block block, Block newState, Player player, CraftingTransaction transaction, Cause cause) {
+        this(block, newState, player, transaction, convert(cause));
+    }
+
+
+    @Since("1.1.1.0-PN")
     public Player getPlayer() {
         return player;
     }
-    
+
+
+    @PowerNukkitOnly
+    @Since("1.1.1.0-PN")
     public CraftingTransaction getTransaction() {
         return transaction;
     }
     
-    public Cause getCause() {
+    @PowerNukkitOnly
+    @Since("1.3.2.0-PN")
+    public DamageCause getDamageCause() {
         return cause;
     }
-    
+
+
+    @Since("1.3.2.0-PN")
+    public int getOldDamage() {
+        return getBlock().getDamage();
+    }
+
+    @Since("1.3.2.0-PN")
+    public int getNewDamage() {
+        return getNewState().getDamage();
+    }
+
+    @Since("1.3.2.0-PN")
+    public void setNewDamage(int newDamage) {
+        getNewState().setDamage(newDamage);
+    }
+
+    @PowerNukkitOnly
+    @Since("1.1.1.0-PN")
+    @Deprecated @DeprecationDetails(
+            since = "1.3.2.0-PN", by = "PowerNukkit",
+            reason = "NukkitX added the class and made getCause() return an enum with a different name.",
+            replaceWith = "getDamageCause()",
+            toBeRemovedAt = "1.5.0.0-PN"
+    )
+    public Cause getCause() {
+        return convert(cause);
+    }
+
+    @PowerNukkitOnly
+    @Since("1.1.1.0-PN")
+    @Deprecated @DeprecationDetails(
+            since = "1.3.2.0-PN", by = "PowerNukkit",
+            reason = "NukkitX added the class but with a different enum for the damage cause",
+            replaceWith = "DamageCause",
+            toBeRemovedAt = "1.5.0.0-PN"
+    )
+    @RequiredArgsConstructor
     public enum Cause {
```

Comment:
Add a way for the plugin developers get the new enum value without having to do switch clauses